### PR TITLE
[IIIF-773] Add VPC, IAM, and Prod environment

### PR DESCRIPTION
**Pull request by cachemeoutside**
The terraform config files do the following
* Setup up a consolidated VPC state
* Setup a consolidated IAM state
* Creates a production EKS cluster with EC2 instances attached
* Creates kubernetes terraform deployments using namespaces associated with endpoint URLs(e.g. prod-iiif, stage-iiif, etc.)

### CODE REVIEW

File: prod/k8s/prod-sinai-iiif/main.tf [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  data = {
+    "tls.crt" = "${base64decode(var.prod_sinai_iiif_tls_crt)}"
+    "tls.key" = "${base64decode(var.prod_sinai_iiif_tls_key)}"
+  }
+
+  type = "kubernetes.io/tls"
+}
+
+module "cantaloupe" {
+  source = "git::https://github.com/UCLALibrary/terraform-kubernetes-cantaloupe.git?ref=IIIF-773"
```
Comment (ksclarke) [inline prod/k8s/prod-sinai-iiif/main.tf:35]: When does this ref get changed?

File: prod/k8s/stage-iiif/main.tf [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  data = {
+    "tls.crt" = "${base64decode(var.stage_iiif_tls_crt)}"
+    "tls.key" = "${base64decode(var.stage_iiif_tls_key)}"
+  }
+
+  type = "kubernetes.io/tls"
+}
+
+module "cantaloupe" {
+  source = "git::https://github.com/UCLALibrary/terraform-kubernetes-cantaloupe.git?ref=IIIF-773"
```
Comment (ksclarke) [inline prod/k8s/stage-iiif/main.tf:35]: When does this ref get changed?

File: prod/k8s/stage-iiif/main.tf [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  cantaloupe_deployment_s3_secret_key = var.cantaloupe_deployment_s3_secret_key
+  cantaloupe_deployment_admin_password = var.cantaloupe_deployment_admin_password
+  cantaloupe_deployment_cpu_limit = var.cantaloupe_deployment_cpu_limit
+  cantaloupe_deployment_cpu_request = var.cantaloupe_deployment_cpu_request
+  cantaloupe_deployment_memory_limit = var.cantaloupe_deployment_memory_limit
+  cantaloupe_deployment_memory_request = var.cantaloupe_deployment_memory_request
+}
+
+module "fester" {
+  source = "git::https://github.com/UCLALibrary/terraform-kubernetes-fester.git?ref=IIIF-773"
```
Comment (ksclarke) [inline prod/k8s/stage-iiif/main.tf:53]: When does this ref get changed?

**Comment by cachemeoutside** [comment]
@ksclarke Ref updated, I also re-did the test cluster structure to use the new Terraform modules.

File: environments/test/k8s/test-iiif/main.tf [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  cantaloupe_deployment_s3_secret_key = var.cantaloupe_deployment_s3_secret_key
+  cantaloupe_deployment_admin_password = var.cantaloupe_deployment_admin_password
+  cantaloupe_deployment_cpu_limit = var.cantaloupe_deployment_cpu_limit
+  cantaloupe_deployment_cpu_request = var.cantaloupe_deployment_cpu_request
+  cantaloupe_deployment_memory_limit = var.cantaloupe_deployment_memory_limit
+  cantaloupe_deployment_memory_request = var.cantaloupe_deployment_memory_request
+}
+
+module "fester" {
+  source = "git::https://github.com/UCLALibrary/terraform-kubernetes-fester.git?ref=IIIF-773"
```
Comment (ksclarke) [inline environments/test/k8s/test-iiif/main.tf:53]: Remove ref?

Comment (cachemeoutside) [reply environments/test/k8s/test-iiif/main.tf]: thanks, i skipped this. resolved.

File: environments/test/k8s/test-iiif/variables.tf [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+variable "cantaloupe_deployment_container_image_url" {
+  type = string
+  default = "uclalibrary/cantaloupe-ucla"
+}
+
+variable "cantaloupe_deployment_container_image_version" {
+  type = string
+  default = "4.1.4"
```
Comment (ksclarke) [inline environments/test/k8s/test-iiif/variables.tf:53]: 4.1.5?

Comment (cachemeoutside) [reply environments/test/k8s/test-iiif/variables.tf]: resolved. thanks!

**Pull request merged**